src/client.ts

```diff
@@ -240,7 +240,6 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     });
 
     this._client = optimizely.createInstance(configWithClientInfo);
-    this.isClientReady = !!configWithClientInfo.datafile;
```

Comment:
I guess this was done to enable synchronous rendering to facilitate SSR. If this is removed, does server side rendering provide correct decision?

Reply:
Its possible to write asynchronous code in a SSR framework. But it must be handled before the component is rendered. Framework like `Next` has this `getServerSideProps` or "Server Components" to handle this scenario.

So we can resolve `onReady` promise before the component is rendered. Thats an option.

Our SDK doc around [SSR](https://github.com/optimizely/react-sdk/tree/master?tab=readme-ov-file#server-side-rendering) seems little confusing to me. 

Now to your point, I am assuming the static check is here for SSR. But how do we make sure that the datafile is valid?

Reply:
Also isn't it a bug in CSR ? It could set the clientReady to true immediately and then later it could be false!!

Reply:
**More context on this.**

We are passing `isServerSide` props in the `OptimizelyProvider` and it has no use in the React Client. We are only using it in hooks like - [useDecision](https://github.com/optimizely/react-sdk/blob/1197fb2483aeb3ba6c85c18d9e605663da73304f/src/hooks.ts#L468). I guess you are talking about this. Here it is checking `isServerSide` and setting clientReady to true immediately. But I guess it still relies on the framework to resolve the `onReady` promise first, and if thats the case then `optimizely?.isReady()` will return proper result. I am just curious about the necessity of `isServerSide` in that case? Maybe I am missing something?

Reply:
**More on this**

Sorry I am dumping lots of things here. I am putting this here for reference. I feel like we need to discuss more on this maybe on a meeting.

[Provider.componentDidUpdate](https://github.com/optimizely/react-sdk/blob/1197fb2483aeb3ba6c85c18d9e605663da73304f/src/Provider.tsx#L94) has a check. SSR does not execute  any lifecycle methods / hooks like - `componentDidMount()`, and `componentDidUpdate()` or `useEffect()` . So this check is invalid.

Reply:
I am not super familiar with all these things myself. As it is used now, isServerSide is probably just same as checking whether a datafile is available synchronously. 

Regarding `isClientReady`, I think what this exactly means is not precise in the current implementation. I think we should rigorously define what it means for the client to be ready in the first place. So this is an area for refactor. If this is set without checking the datafile validity, it can later become false. So yeah, the removed looks wrong. I don't have any issue removing this, I just wanted to ensure that SSR render works when we remove this. If it does, we can go ahead and remove this.

Let's discuss it further in a call.

Reply:
Cool. I will check if everything works as expected in SSR with / without it before our 1:1